### DISCUSSION TITLE
how to rollback

### DISCUSSION THREAD

**Discussion by SpringHgui** [kubernetes-client/csharp#1105] [Q&A]
how to do [rollback-daemon-set](https://kubernetes.io/docs/tasks/manage-daemon/rollback-daemon-set/) by sdk

**Comment by brendandburns** [comment]
Rollback is moderately complicated collection of API calls, there's no one call that directly maps to 'rollback'

You can see the Go code here:
https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/rollout/rollout_undo.go#L161

It's a little thorny to follow, but the daemonset version is here:
https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollback.go#L263

You should be able to emulate this by performing a patch just like the go client does:
https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollback.go#L306

But you will need to extract the history from the DaemonSet (you'll need to `GET` it first)

If you do get this code together, we'd be happy to take a PR to support this.

**Comment by SpringHgui** [comment]
thank you for  support `rollback daemonset and deploymnet`

**Comment by tg123** [comment]
project https://github.com/orgs/kubernetes-client/projects/1

**Reply by SpringHgui** [reply]
this link got 404

**Reply by tg123** [reply]
omg, i cant set access

basically, you will have all kubectl experience in c# sdk.
too busy to finish it, but will be working on it soon

**Reply by SpringHgui** [reply]
very nice~
